indico/web/client/js/react/components/FileSubmission.jsx

```diff
@@ -0,0 +1,113 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useCallback, useRef, useState} from 'react';
+import PropTypes from 'prop-types';
+import Dropzone from 'react-dropzone';
+import {Button, Card, Divider, Grid, Header, Icon, Segment} from 'semantic-ui-react';
+
+import {TooltipIfTruncated} from 'indico/react/components';
+import {Translate} from 'indico/react/i18n';
+
+import './FileSubmission.module.scss';
+
+export default function FileSubmission({onChange}) {
+  const [files, setFiles] = useState([]);
+  const onDrop = useCallback(
+    acceptedFiles => {
+      setFiles(acceptedFiles);
+      if (onChange) {
+        onChange(acceptedFiles);
+      }
+    },
+    [setFiles, onChange]
+  );
+
+  const removeFile = file => {
+    const newFiles = files.filter(f => !Object.is(f, file));
+    setFiles(newFiles);
+    if (onChange) {
+      onChange(newFiles);
+    }
+  };
+
+  const dropzoneRef = useRef();
+  const openFileDialog = () => {
+    if (dropzoneRef.current) {
+      dropzoneRef.current.open();
+    }
+  };
+
+  return (
+    <Dropzone ref={dropzoneRef} onDrop={onDrop} multiple noClick noKeyboard>
+      {({getRootProps, getInputProps, isDragActive}) => (
+        <div {...getRootProps()} styleName="dropzone-area">
+          <input {...getInputProps()} />
+          <Segment textAlign="center" placeholder>
+            <Grid celled="internally">
+              <Grid.Row columns={files.length === 0 ? 1 : 2}>
+                {!isDragActive && files.length !== 0 && (
+                  <Grid.Column width={10} verticalAlign="middle">
+                    <Card.Group itemsPerRow={files.length === 1 ? 1 : 2} centered>
+                      {files.map(file => (
+                        <Card
+                          styleName="file-card"
+                          key={file.name}
+                          centered={files.length === 1}
+                          raised
+                        >
+                          <Icon name="trash" onClick={() => removeFile(file)} color="red" />
+                          <Card.Content>
+                            <TooltipIfTruncated>
+                              <Card.Header textAlign="center">{file.name}</Card.Header>
+                            </TooltipIfTruncated>
+                            <Card.Meta textAlign="center">{`${(file.size / 1024).toFixed(
+                              2
+                            )} KB`}</Card.Meta>
```

Comment:
It's `kB`.

Eventually we should add a util that formats a file size nicely (using the most suitable unit)